subprojects/test-runner/report-viewer/src/main/kotlin/com/avito/report/internal/ReportsAddApiImpl.kt

```diff
@@ -136,7 +137,13 @@ internal class ReportsAddApiImpl(private val client: JsonRpcClient) : ReportsAdd
         val description = test.description
         if (!description.isNullOrBlank()) report["description"] = description
 
-        if (video != null) report["video"] = video
+        if (video != null) {
+            when (val fileAddress = video.link) {
```

Comment:
Visual change, see discussion [here](http://links.k.avito.ru/8c)

Reply:
I think the key problem is that `report` is a fucking `mutableMap`
Let's change `report` to an object with `type` contract and make a serializer for that type